cli-deploy-aws-s3.js

```diff
@@ -0,0 +1,143 @@
+
+// #region Commands
+
+const deployAwsS3 = {
+
+    cmd(src, dest) {
+        return `aws s3 sync ${src} s3://${dest}`;
+    },
+    description() {
+        return "Publish build artifacts to Amazon S3 storage";
+    },
+    run() {
+        // Check system/command requirements
+        this.validateOrExit();
+
+        // Locally publish frontend via webpack
+        if (program.publish && program.webpack) {
+            const publishResult = webpackPublish.run();
+            if (!publishResult) {
+                shell.exit(1);
+            }
+        }
+
+        // Deploy build artifacts to S3
+        echo.message("Copying local build artifacts to Amazon S3...");
+        echo.message(` - Profile: ${profile}`);
+        echo.message(` - Source path: ${sourcePath}`);
+        echo.message(` - Destination path: ${destination}`);
+
+        const syncCommand = this.cmd(sourcePath, destination) + ` --profile ${profile}`;
+        echo.message(` - Command: ${syncCommand}`);
+        if (shell.exec(syncCommand, { silent: false }).code !== 0) {
+            echo.error(" - Failed to deploy to AWS S3");
+            shell.exit(1);
+        }
+
+        echo.newLine();
+        echo.success("Application successfully deployed to AWS S3!");
+    },
+    validateOrExit() {
+        const errors = [];
+
+        // Validate arguments
+        profile = program.profile;
+        if (profile === undefined || profile === null) {
+            errors.push("--profile is required");
+        }
+
+        destination = program.destination;
+        if (destination === undefined || destination === null) {
+            errors.push("--destination is required");
+        }
+
+        if (program.source !== undefined && program.source !== null) {
+            sourcePath = program.source;
+        }
+
+        // Bail if up-front arguments are errored
+        if (errors.length > 0) {
+            echo.errors(errors);
+            shell.exit(1);
+        }
+
+        if (!shell.which("python")) {
+            echo.error(`Python 3.7+ is required - ${pythonInstallerUrl}`);
+            shell.exit(1);
+        }
+
+        if (!shell.which("pip")) {
+            echo.error(`PIP is required - ${pythonInstallerUrl}`);
+            shell.exit(1);
+        }
+
+        if (!shell.which("aws")) {
+            echo.message("AWS CLI not found. Installing via PIP...");
+
+            if (shell.exec("pip install awscli").code !== 0) {
+                echo.error("Failed to install aws cli via pip");
+                shell.exit(1);
+            }
+
+            echo.success(" - Successfully installed AWS CLI");
+        }
+
+        // Handle errors
+        if (errors.length > 0) {
+            echo.errors(errors);
+            shell.exit(1);
+        }
+
+        return true;
+    },
+};
+
+// #endregion Commands
+
+
+/**************************************************************************************************
+ * Entrypoint / Command router
```

Comment:
Do we need this comment as well as the #region? It kinda feels like they both serve the same purpose, to differentiate the block of code, so it feels redundant to me.

Reply:
Its a compromise honestly. I don't personally leverage #regions like this and find code more readable with the block comments.